app/controllers/providers_controller.rb

```diff
@@ -1,4 +1,6 @@
 class ProvidersController < ApplicationController
+  rescue_from JsonApiClient::Errors::NotFound, with: :not_found
```

Comment:
Similar to mc-be, I think we should have a before-hook setter that finds `@provider` and returns an appropriate error when it's not found.

Reply:
We tried this and it didnt work so we put it at controller level